sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableAccountSasGenerator.java

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables.implementation;
+
+import com.azure.core.credential.AzureNamedKeyCredential;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.tables.sas.TableAccountSasPermission;
+import com.azure.data.tables.sas.TableAccountSasSignatureValues;
+import com.azure.data.tables.sas.TableSasIpRange;
+import com.azure.data.tables.sas.TableSasProtocol;
+
+import java.time.OffsetDateTime;
+import java.util.Objects;
+
+import static com.azure.data.tables.implementation.TableSasUtils.computeHMac256;
+import static com.azure.data.tables.implementation.TableSasUtils.formatQueryParameterDate;
+import static com.azure.data.tables.implementation.TableSasUtils.logStringToSign;
+import static com.azure.data.tables.implementation.TableSasUtils.tryAppendQueryParameter;
+
+/**
+ * A class containing utility methods for generating SAS tokens for the Azure Storage accounts.
+ */
+public class TableAccountSasGenerator {
+    private final ClientLogger logger = new ClientLogger(TableAccountSasGenerator.class);
+    private String version;
+    private TableSasProtocol protocol;
+    private OffsetDateTime startTime;
+    private OffsetDateTime expiryTime;
+    private String permissions;
+    private TableSasIpRange sasIpRange;
+    private String services;
+    private String resourceTypes;
+
+    /**
+     * Creates a new {@link TableAccountSasGenerator} with the specified parameters.
+     *
+     * @param sasValues The {@link TableAccountSasSignatureValues account signature values}.
+     */
+    public TableAccountSasGenerator(TableAccountSasSignatureValues sasValues) {
+        this.version = sasValues.getVersion();
+        this.protocol = sasValues.getProtocol();
+        this.startTime = sasValues.getStartTime();
+        this.expiryTime = sasValues.getExpiryTime();
+        this.permissions = sasValues.getPermissions();
+        this.sasIpRange = sasValues.getSasIpRange();
+        this.services = sasValues.getServices();
+        this.resourceTypes = sasValues.getResourceTypes();
+    }
+
+    /**
+     * Generates a Sas signed with a {@link AzureNamedKeyCredential}.
+     *
+     * @param azureNamedKeyCredential {@link AzureNamedKeyCredential}
+     * @param context Additional context that is passed through the code when generating a SAS.
+     * @return A String representing the Sas
+     */
+    public String generateSas(AzureNamedKeyCredential azureNamedKeyCredential, Context context) {
```

Comment:
Sure, we can do that. I won't look exactly like what we have for other storage classes used to generate a SAS but I think kit's ok because it's an implementation detail.